sympy/core/mul.py

```diff
@@ -275,6 +275,7 @@ def flatten(cls, seq):
 
         from sympy.calculus.accumulationbounds import AccumBounds
         from sympy.matrices.expressions import MatrixExpr
+        from sympy.physics.units import Dimension
```

Comment:
It is possible that importing units here affects the benchmark timings. That might be an incorrect measurement in some sense because it measures the cost of the import. I have seen other cases where a benchmark reported a slowdown that was actually just caused by importing the units module because it is slow to import.

Note that the units module is deliberately not imported at sympy start up because it is slow but this will cause it to be imported.